src/contracts/libraries/BN254.sol

```diff
@@ -139,12 +139,20 @@ library BN254 {
      */ 
     function scalar_mul_tiny(BN254.G1Point memory p, uint16 s) internal view returns (BN254.G1Point memory) {
         require(s < 2**9, "scalar-too-large");
+
+        uint256 n = 1;
+        uint256 m = 1;
+        while(s > m){
```

Comment:
this is close to what i was thinking, but couldn't we fit this inside the main loop? or would that make the gas larger for some reason?

also could we have a special case for `s == 1`? to avoid the doubling at the end?